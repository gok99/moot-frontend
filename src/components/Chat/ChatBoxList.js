import React from 'react';
import { Col } from 'react-bootstrap';
import ChatBox from './ChatBox';

import '../Styles/styles.css';

function ChatBoxList(props) {
  const chats = props.chatList;
  const chatsList = chats.map((chat, index) => {
    return <a href={"https://t.me/moot_chat"+(index+1)+"_bot"} target="_blank" rel="noopener noreferrer" className="blocklink">
            <ChatBox chatboxId={index+1} activeStatus={chat.active} activeMatchUUID={chat.activematchUUID}></ChatBox>
          </a>
  });
  return (
    <Col>{chatsList}</Col>
  );
}

export default ChatBoxList;