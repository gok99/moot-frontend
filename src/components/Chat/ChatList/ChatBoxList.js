import React, { useRef } from 'react';
import { Row, Col, Button, OverlayTrigger, Popover } from 'react-bootstrap';
import ChatBox from './ChatBox';

import '../../Styles/styles.css';
import '../chat.css';
import { findByAltText } from '@testing-library/react';

const ChatBoxList = (props) => {
  const chats = props.chatList;
  const chatInfo = props.chatInfo;
  const ref = useRef(null);
  const chatsList = chats.length === 0 
    ? null 
    : chats.map((chat, index) => {
        const description = chatInfo["chat" + (index + 1)].description;
        const tags = chatInfo["chat" + (index + 1)].tags;
        const str = tags.length <= 0 ? "No declared interests" : tags.slice(1).reduce((acc, curr) => `${acc}, ${curr}`, tags[0]);
        return <Row className="mb-2">
                  <Col md={9}>
                    <OverlayTrigger
                      trigger="click"
                      key="bottom"
                      placement="bottom"
                      container={ref.current}
                      overlay={
                        <Popover className="popover-chat" id={`popover-positioned-bottom`}>
                          <Popover.Title as="h3">{chat.active ? "You have an ongoing match!" : "This match slot is empty..."}</Popover.Title>
                          { chat.active
                            ? <>
                                <Popover.Content>
                                  This chat is anonymous, but here's what we can tell you about your match:<br /><br />
                                  <strong>Description</strong>: <br />{description}<br /><br />
                                  <strong>Interests</strong>: <br />{str}
                                </Popover.Content>
                              </>
                            : <Popover.Content>
                                Use the <strong>Match</strong> or <strong>QuickMatch</strong> features to match up with another anonymous user!
                              </Popover.Content>
                          }
                        </Popover>
                      }
                    >
                      <Button ref={ref} type="button" className="btn-chatbox">
                        <ChatBox chatId={index+1} status={chat.active}></ChatBox>
                      </Button>
                    </OverlayTrigger>
                  </Col>
                  <Col md={3} className="d-flex justify-content-md-center">
                    <Button className="d-flex justify-content-center btn-chatlink text-chatboxheader" href={"https://t.me/moot_chat"+(index+1)+"_bot"} target="_blank" rel="noopener noreferrer">Chat #{index+1} &gt;&gt;&gt;</Button>
                  </Col>
                </Row>
      });
  return (
    <Col>
      <h2>My Chats</h2>
      <hr />
      {chatsList}
    </Col>
  );
};

export default ChatBoxList;