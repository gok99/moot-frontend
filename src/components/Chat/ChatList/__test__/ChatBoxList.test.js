import React from 'react';
import ReactDOM from 'react-dom';
import ChatBoxList from '../ChatBoxList';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatBoxList chatList={[]} chatInfo={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<ChatBoxList chatList={[]} chatInfo={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});
