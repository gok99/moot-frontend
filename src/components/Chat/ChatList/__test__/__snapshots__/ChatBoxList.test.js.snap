// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col"
>
  <h2>
    My Chats
  </h2>
  <hr />
</div>
`;
